src/lib.rs

```diff
@@ -241,14 +251,17 @@ pub trait ForumUserRegistry<AccountId> {
 }
 
 /// Convenient composite time stamp 
-struct BlockchainTimestamp<BlockNumber, Moment> {
+//#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq)]
+pub struct BlockchainTimestamp<BlockNumber, Moment> {
```

Comment:
This could be simplified. We already have a similar structure in the storage module.
See how it is used here: https://github.com/Joystream/substrate-runtime-joystream/blob/ae76fc45109562ddb9415f0e333527b2b773a01d/src/storage/data_directory.rs#L44
```suggestion
pub struct BlockchainTimestamp<T: Trait> {
    block: T::BlockNumber,
    time: T::Moment,
}
```

Reply:
This is generally not the approach taken in this module, we don't want to require polution of the associated types on the runtime, so explicit generics are used. This done across this entire runtime, not just`BlockchainTimestamp`.